src/nvim/ex_cmds.c

```diff
@@ -5846,3 +5904,307 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
+    curbuf->b_p_ma = false;  // Not Modifiable
+    curwin->w_p_fen = false;
+    curwin->w_p_rl = cmdmsg_rl;
+    cmdmsg_rl = false;
+    RESET_BINDING(curwin);
+
+    // Do execute autocommands for setting the filetype (load syntax).
+    unblock_autocmds();
+
+    // Showing the prompt may have set need_wait_return, reset it.
+    need_wait_return = false;
+
+    // Reset 'textwidth' after setting 'filetype'
+    // (the Vim filetype plugin sets 'textwidth' to 78).
+    curbuf->b_p_tw = 0;
+
+    // Save the buffer used in the split
+    livebuf = curbuf;
+
+    // Initialize line and highlight variables
+    int line = 0;
+    int src_id_highlight = 0;
+    long sub_size = STRLEN(sub);
+    long pat_size = STRLEN(pat);
+
+    // Get the width of the column which display the number of the line
+    long highest_num_line = 0;
+    kl_iter(matchedline_T, lmatch, current)
+      highest_num_line = (*current)->data.lnum;
+
+    // computing the length of the column that will display line number
+    int col_width = log10(highest_num_line) + 1 + 3;
+
+    // allocate a line sized for the window
+    char *str = xcalloc((size_t) curwin->w_frame->fr_width, sizeof(char));
+
+    // Append the lines to our buffer
+    kl_iter(matchedline_T, lmatch, current) {
+      matchedline_T mat = (*current)->data;
+      size_t line_size = STRLEN(mat.line) + col_width + 1;
+
+      // Reallocation if str not long enough
+      if (line_size > curwin->w_frame->fr_width * sizeof(char)) {
+        str = xrealloc(str, line_size * sizeof(char));
+      }
+
+      // Add the line number to the string
+      char *col = compute_line_number(col_width, mat.lnum);
+      snprintf(str, line_size, "%s%s", col, mat.line);
+      ml_append(line++, (char_u *) str, (colnr_T) 0, false);
+
+      int i = 0;
+      kl_iter(colnr_T, mat.start_col, col) {
+        // highlight the replaced part
+        if (sub_size > 0) {
+          src_id_highlight =
+            bufhl_add_hl(curbuf,
+                         src_id_highlight,
+                         2,  // id of our highlight
+                         line,
+                         (*col)->data + col_width + i * (sub_size - pat_size) + 1,
+                         (*col)->data + col_width + i * (sub_size - pat_size) +
+                         sub_size);
+        }
+        i++;
+      }
+
+      // free of the saved line and the allocated column
+      xfree(col);
+      xfree((*current)->data.line);
+    }
+    xfree(str);
+  }
+
+  redraw_later(SOME_VALID);
+
+  // Restore the old window
+  win_enter(oldwin, false);
+  finish_live_cmd(save_State, &winsizes, save_exmode,
+                  save_restart_edit, save_cmdmsg_rl, 1);
+
+  return;
+}
+
+/// Parse the substitution command line
+//
+/// @param eap arguments of the substitution
+/// @return cmdl_progress
+/// @see LiveSub_state definition
+LiveSub_state parse_sub_cmd(exarg_T *eap) {
+  int i = 0;
+  LiveSub_state cmdl_progress;
+
+  if (eap->arg[i++] != '/') {
```

Comment:
this is not completely general, vim allows some other delimiters than `/`, for instance `:%s+/+//+g` to double every forward slash.